backend/dataall/modules/s3_datasets/db/dataset_repositories.py

```diff
@@ -202,6 +202,15 @@ def list_all_datasets(session) -> [S3Dataset]:
     def list_all_active_datasets(session) -> [S3Dataset]:
         return session.query(S3Dataset).filter(S3Dataset.deleted.is_(None)).all()
 
+    @staticmethod
+    def list_all_active_datasets_with_glue_db(session, dataset_uri) -> [S3Dataset]:
+        s3_dataset: S3Dataset = DatasetRepository.get_dataset_by_uri(session=session, dataset_uri=dataset_uri)
+        return (
+            session.query(S3Dataset)
+            .filter(and_(S3Dataset.deleted.is_(None), S3Dataset.GlueDatabaseName == s3_dataset.GlueDatabaseName))
```

Comment:
As you pointed out in your testing scenario 3, there might be issues when 2 datasets from different environments share the same glue database name